src/Common/src/System/Diagnostics/Debug.cs

```diff
@@ -84,19 +144,33 @@ public static void WriteLine(string message)
         [System.Diagnostics.Conditional("DEBUG")]
         public static void Write(string message)
         {
-            s_logger.WriteCore(message ?? string.Empty);
+            if (message == null)
+            {
+                WriteCore(string.Empty);
+                return;
+            }
+            if (s_needIndent)
+            {
+                message = GetIndentString() + message;
+                s_needIndent = false;
+            }
+            WriteCore(message);
+            if (message.EndsWith(Environment.NewLine))
+            {
+                s_needIndent = true;
```

Comment:
What happens when Debug is used by multiple threads concurrently?  They're all sharing the same s_needIndent static, there's no locking, etc.